pkg/schema/schema_test.go

```diff
@@ -0,0 +1,554 @@
+// Copyright 2023 The Okteto Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package schema
+
+import (
+	"strings"
+	"testing"
+
+	"github.com/santhosh-tekuri/jsonschema/v6"
+	"github.com/stretchr/testify/assert"
+)
+
+func validateOktetoManifest(content string) error {
+	oktetoJsonSchema, err := NewJsonSchema().ToJSON()
+	if err != nil {
+		return err
+	}
+
+	var obj interface{}
+	err = unmarshal([]byte(content), &obj)
+	if err != nil {
+		return err
+	}
+
+	compiler := jsonschema.NewCompiler()
+	doc, err := jsonschema.UnmarshalJSON(strings.NewReader(string(oktetoJsonSchema)))
+	if err != nil {
+		return err
+	}
+
+	resourceName := "schema.json"
+
+	err = compiler.AddResource(resourceName, doc)
+	if err != nil {
+		return err
+	}
+
+	schema, err := compiler.Compile(resourceName)
+	if err != nil {
+		return err
+	}
+
+	err = schema.Validate(obj)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func Test_Schema(t *testing.T) {
```

Comment:
This test mentions other repos for sake of using real manifests, but I it could be using other manifests with different test names. The purpose is solely to have real manifests to be tested as whole, instead of testing each section invididualy.

It's not intended as integration test, but should be kept as unit and be used to validate changes in the schema pkg.